### DISCUSSION TITLE
Questions on Power Consumption / Deep Sleep

### DISCUSSION THREAD

**Discussion by jenspr** [HomeSpan/HomeSpan#250] [Q&A] [answered]
First post here, so apologize if asking stupid question. 

I have had a Homekit device running on ESP8266. It had to be connected to WIFI all the time. There was no way to go to sleep or power saving as - at least that is what I understood - Homekit expected the device to always be connected/active.
So battery powered no option at all :-(

As Homespan on ESP32 offers Bluettooth as an option, I wonder what the chances are here.
Can one build it in a way to allow lower power usage or even sleep and still have it operate on Homekit?

I wonder how other (off-the-shelf) products do that. Are they using different microcontroller do all other do rely on always-on micorcontroller/connection.

I tried to search on the Internet, but did not find good answers. 
I am embarking on some projects with homespan right now.. so I am curious :-)

**Comment by HomeSpan** [comment, marked as answer]
Apple HomeKit works with both WiFi and Bluetooth.  However, the protocols are quite different and most devices use WiFi for reliability (and perhaps range).  Though the ESP32 has Bluetooth, HomeSpan has not implemented the HomeKit HAP-R2 Bluetooth protocols and only works with WiFi (it would be a  significant project to implement the HomeKit Bluetooth protocol in HomeSpan).

As you note, WiFi is not practical under battery power, especially since HomeKit requires a permanent, active connection, to every device, which typically precludes the use of sleep-modes or power-save modes to save battery life.

Many battery-powered HomeKit devices use a powered base-hub connected to WiFi (or your local Ethernet) with remote sensors or controls connected to the base hub using either Bluetooth or some other proprietary method.  For example, Lutron Caseta Pico switches are battery powered, seem to last forever, and work with HomeKit.  But that's because they communicate via a proprietary protocol to a powered Lutron Caseta Base Station, and it is the Base Station that is actually connected to HomeKit.

**Reply by jenspr** [reply]
Thank you very much.
So there is no actual good (home-brew) solution for Homekit that could run well on battery I got from this. 
Then will stick with this library and WIFI for a while. :-) Thanks

**Reply by yrjyrj123** [reply]
Is HomeKit Bluetooth protocol implementation already in the works or planned?
I really need this feature.
I would be grateful if you could add it.

**Reply by HomeSpan** [reply]
No plans to add Bluetooth to HomeSpan.
